operators/src/source/gdal_source.rs

```diff
@@ -0,0 +1,456 @@
+        let time_interval_iterator = self.time_interval_provider.clone().into_iter();
+        time_interval_iterator.flat_map(
+            move |time| 
+                self.grid_tile_provider.tile_informations().into_iter().map(
+                move |tile| (time, tile)
+            )
+        
+        )
+    }
+
+    ///
+    /// A method to load single tiles from a GDAL dataset.
+    ///
+    pub fn load_tile_data<T: gdal::raster::types::GdalType + Copy>(&self, time_interval: &TimeInterval, tile_information: &TileInformation) -> Result<RasterTile2D<T>> {
+            
+        // snap the time interval start instance to the dataset tick configuration
+        let time_string = self.gdal_params.tick.map(
+            |t| t.snap_datetime(
+                &time_interval.start().as_naive_date_time()
+            ).format(&self.gdal_params.time_format).to_string()
+        );
+        
+        let file_name = self.gdal_params.file_name_with_time_placeholder.replace("%%%_START_TIME_%%%", &time_string.unwrap_or_else(|| "".into()));
+        
+        let path = self.gdal_params.base_path.join(&file_name);
+
+        // open the dataset at path (or 'throw' an error)
+        let dataset = GdalDataset::open(&path).expect("cant open dataset");
+        // get the geo transform (pixel size ...) of the dataset (or 'throw' an error)
+        let gdal_geo_transform = dataset.geo_transform()?;
+        let geo_transform = GeoTransform::from(gdal_geo_transform);
+        // TODO: clip the geotransform information?
+
+        // get the requested raster band of the dataset …
+        let rasterband_index = self.gdal_params.channel.unwrap_or(1) as isize; // TODO: investigate if this should be isize in gdal
+        let rasterband: GdalRasterBand = dataset.rasterband(rasterband_index)?;
+                    
+        let &[.., y_pixel_position, x_pixel_position] = tile_information.global_pixel_position().dimension_size();
+        let &[.., y_tile_size, x_tile_size] = tile_information.tile_size_in_pixels.dimension_size();
+        // read the data from the rasterband
+        let pixel_origin = (x_pixel_position as isize, y_pixel_position as isize);
+        let pixel_size = (x_tile_size, y_tile_size);
+        let buffer = rasterband.read_as::<T>(pixel_origin, // pixelspace origin
+            pixel_size, // pixelspace size
+            pixel_size /* requested raster size */
+        )?;
+        // println!("pixel_origin: {:?}, pixel_size: {:?}, size: {:?}", pixel_origin, pixel_size, pixel_size);
+
+        // TODO: get the raster metadata!
+
+        let raster_result = Raster2D::new(tile_information.tile_size_in_pixels, buffer.data, None, *time_interval, geo_transform)?;
+        Ok(RasterTile2D::new(*time_interval, tile_information.clone(), raster_result))
+    }
+
+    ///
+    /// A stream of `RasterTile2D`
+    ///
+    pub fn tile_stream<T: gdal::raster::types::GdalType + Copy>(&self) -> BoxStream<Result<RasterTile2D<T>>> {
```

Comment:
Isn't it a `TryStream` then?

Reply:
We need to discuss this. A trystream is not a stream of results

Reply:
> A convenience for streams that return Result values that includes a variety of adapters tailored to such futures.

I thought that it is. But I also saw that I myself changed it back to `BoxStream` for csv because it was strange to work with it.